migrator/src/test/java/com/google/phonenumbers/migrator/MetadataZipFileReaderTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2020 The Libphonenumber Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.phonenumbers.migrator;
+
+import static com.google.common.truth.Truth8.assertThat;
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.phonenumbers.migrator.testing.testUtils.AssertUtil.assertThrows;
+
+import com.google.common.truth.Truth;
+import com.google.i18n.phonenumbers.metadata.DigitSequence;
+import com.google.i18n.phonenumbers.metadata.table.CsvTable;
+import com.google.i18n.phonenumbers.metadata.table.RangeKey;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Paths;
+import java.util.Optional;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class MetadataZipFileReaderTest {
+  private static final String TEST_DATA_PATH = "./src/test/java/com/google/phonenumbers/migrator/testing/testData/";
+
+  @Test
+  public void testInvalidFileLocation() {
+    String fileLocation = "invalid-zipfile-location";
+      try {
+        MetadataZipFileReader.of(Paths.get(fileLocation));
+        Assert.fail("Expected IOException and did not receive");
```

Comment:
You could also use assertThrows(NoSuchFileException.class, () -> MetadataZipFileReader.of(Paths.get(fileLocation)));

Reply:
Discussed offline. This is not possible with junit4 and our current implementation uses Runnable, where the 'run' method doesn't declare an exception so we can leave this as is.